### PR TITLE
Battle-Engine: Show all types of Pokeballs

### PR DESCRIPTION
**Pull request by Lord-Haji**

### CODE REVIEW

File: battle-engine.js [hunk 1]
```diff
@@ -3466,7 +3466,13 @@ class Battle extends Tools.BattleDex {
 
 		this.makeRequest('move');
 
-		if (Tools.getFormat(this.format).isWildEncounter) this.add('raw', '<button name="send" value="/throwpokeball pokeball" class="button">Throw a pokeball</button>');
+		if (Tools.getFormat(this.format).isWildEncounter) {
+			 this.add('raw', '<h3>Throw a Pokéball !!</h3>')
```
Comment (HoeenCoder) [inline battle-engine.js]: The buttons are self-explantiory no need to add a message about them

File: battle-engine.js [hunk 2]
```diff
@@ -3466,7 +3466,13 @@ class Battle extends Tools.BattleDex {
...
+			 let balls = ['pokeball', 'greatball', 'ultraball', 'masterball'];
+			 for (let i = 0; i <= balls.length; i++) {
+			 		this.add('raw', '<button name="send" value="/throwpokeball "' + balls[i] + '" style="background:transparent;border:none;"><img src="http://www.serebii.net/itemdex/sprites/pgl/' + balls[i] + '.png" width="30" height="30"></button>')
```
Comment (HoeenCoder) [inline battle-engine.js]: Add all the buttons in a single message to prevent spam

Comment (Lord-Haji) [reply battle-engine.js]: there is no <br> it will be on same line ??

File: battle-engine.js [hunk 3]
```diff
@@ -3466,7 +3466,12 @@ class Battle extends Tools.BattleDex {
 
 		this.makeRequest('move');
 
-		if (Tools.getFormat(this.format).isWildEncounter) this.add('raw', '<button name="send" value="/throwpokeball pokeball" class="button">Throw a pokeball</button>');
+		if (Tools.getFormat(this.format).isWildEncounter) {
+			 let balls = ['pokeball', 'greatball', 'ultraball', 'masterball'];
+			 for (let i = 0; i <= balls.length; i++) {
+			 	this.add('raw', '<button name="send" value="/throwpokeball "' + balls[i] + '" style="background:transparent;border:none;"><img src="http://www.serebii.net/itemdex/sprites/pgl/' + balls[i] + '.png" width="30" height="30"></button>');
```
Comment (HoeenCoder) [inline battle-engine.js]: this is still adding 4 messages instead of 1 ._.

File: battle-engine.js [hunk 4]
```diff
@@ -3466,7 +3466,12 @@ class Battle extends Tools.BattleDex {
 
 		this.makeRequest('move');
 
-		if (Tools.getFormat(this.format).isWildEncounter) this.add('raw', '<button name="send" value="/throwpokeball pokeball" class="button">Throw a pokeball</button>');
+		if (Tools.getFormat(this.format).isWildEncounter) {
+			 let balls = ['pokeball', 'greatball', 'ultraball', 'masterball'];
+			 balls.forEach(pokeball => {
+				this.add('raw', '<button name="send" value="/throwpokeball ' + pokeball + '" style="background:transparent;border:none;"><img src="http://www.serebii.net/itemdex/sprites/pgl/' + pokeball + '.png" width="30" height="30"></button>&nbsp;&nbsp;');
+			});
```
Comment (HoeenCoder) [inline battle-engine.js]: Ive told you 3 times now, send all the buttons in 1 message like this:
```javascript
let buttons = '';
for (let i = 0; i < balls.length; i++) {
  buttons += '<button name="send" value="/throwpokeball ' + balls[i] + '" style="background:transparent;border:none;"><img src="http://www.serebii.net/itemdex/sprites/pgl/' + balls[i] + '.png" width="30" height="30"></button>&nbsp;&nbsp;';
}
this.add('raw', buttons);
```
Also don't use `forEach` on arrays, it has bad performance compared with other options like `for of` or a normal loop.

File: battle-engine.js [hunk 5]
```diff
@@ -3466,7 +3466,14 @@ class Battle extends Tools.BattleDex {
...
 		this.makeRequest('move');
 
-		if (Tools.getFormat(this.format).isWildEncounter) this.add('raw', '<button name="send" value="/throwpokeball pokeball" class="button">Throw a pokeball</button>');
+		if (Tools.getFormat(this.format).isWildEncounter) {
+			let balls = ['pokeball', 'greatball', 'ultraball', 'masterball'];
+			let buttons = '';
+			for (let i = 0; i < balls.length; i++) {
+				buttons += '<button name="send" value="/throwpokeball ' + balls[i] + '" style="background:transparent;border:none;"><img src="http://www.serebii.net/itemdex/sprites/pgl/' + balls[i] + '.png" width="30" height="30"></button>&nbsp;&nbsp;';
+			}
+			this.add('raw', buttons);
```
Comment (HoeenCoder) [inline battle-engine.js:3475]: add ``this.add('');`` so the on battle screen message bar clears

Comment (Lord-Haji) [reply battle-engine.js:3475]: i couldnt understand

Comment (HoeenCoder) [reply battle-engine.js:3475]: Add that after the ``this.add('raw', buttons);``

**Pull request merged**